backend/shared/shared/oidc/views/hki_views.py

```diff
@@ -57,38 +56,47 @@ def login_failure(self):
         return HttpResponseRedirect(url)
 
 
-class HelsinkiOIDCLogoutView(OIDCLogoutView):
-    """Override OIDCLogoutView to match the keycloak backchannel logout"""
+class HelsinkiOIDCLogoutView(View):
+    """
+    Initiate logout process with Keycloak.
+    Use GET request like e.g. Django auth logout does."""
 
-    def post(self, request):
+    http_method_names = ["get"]
+
+    def get(self, request):
         if request.user.is_authenticated:
             if not request.session.get("oidc_id_token"):
                 auth.logout(request)
-                return HttpResponse("OK", status=200)
+                return HttpResponseRedirect(settings.LOGOUT_REDIRECT_URL)
 
-            payload = {
+            params = {
                 "id_token_hint": request.session.get("oidc_id_token"),
-                "refresh_token": request.session.get("oidc_refresh_token"),
-                "client_id": settings.OIDC_RP_CLIENT_ID,
-                "client_secret": settings.OIDC_RP_CLIENT_SECRET,
+                "post_logout_redirect_uri": settings.OIDC_OP_LOGOUT_CALLBACK_URL,
             }
 
-            response = requests.post(
-                settings.OIDC_OP_LOGOUT_ENDPOINT,
-                data=payload,
-                verify=self.get_settings("OIDC_VERIFY_SSL", True),
-                timeout=self.get_settings("OIDC_TIMEOUT", None),
-                proxies=self.get_settings("OIDC_PROXY", None),
+            query = urlencode(params)
+
+            redirect_url = "{url}?{query}".format(
+                url=settings.OIDC_OP_LOGOUT_ENDPOINT, query=query
```

Comment:
Would using urllib.parse.urljoin or some other URL handling function here be safer than using format?

Reply:
fixed